features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/metric/SessionEndedMetric.kt

```diff
@@ -0,0 +1,222 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.metric
+
+import com.datadog.android.core.metrics.PerformanceMetric
+import com.datadog.android.rum.internal.domain.scope.RumViewManagerScope
+import com.datadog.android.rum.model.ViewEvent
+import java.util.concurrent.TimeUnit
+
+/**
+ * Metric for rum session ended event.
+ */
+@Suppress("TooManyFunctions")
+internal class SessionEndedMetric(
+    private val sessionId: String,
+    private val precondition: String
+) {
+
+    private val trackedViewMap by lazy { mutableMapOf<String, TrackedView>() }
+
+    private val trackedErrorMap by lazy { mutableMapOf<String, Int>() }
```

Comment:
minor: maybe it makes sense to change the naming

```suggestion
    private val trackedViewsById by lazy { mutableMapOf<String, TrackedView>() }

    private val errorKindFrequencies by lazy { mutableMapOf<String, Int>() }
```

Reply:
updated